polygon/bor/bordb/prune.go

```diff
@@ -97,13 +97,13 @@ func UnwindHeimdall(ctx context.Context, heimdallStore heimdall.Store, bridgeSto
 		}
 	}
 
-	if !unwindCfg.KeepEventNums {
+	if !unwindCfg.KeepEventNums && unwindCfg.Astrid {
```

Comment:
should not be needed, only `bridge.UnwindEventProcessedBlocks` and `UnwindSpanBlockProducerSelections` are guarded by the `Astrid` flag as per https://github.com/erigontech/erigon/pull/12148/files